app/scripts/controllers/transactions/index.test.js

```diff
@@ -3426,22 +3426,44 @@ describe('Transaction Controller', function () {
       });
     });
 
-    it('ignores new transactions if hash matches existing transaction', async function () {
-      const existingTransaction = TRANSACTION_META_MOCK;
-      const incomingTransaction1 = { ...TRANSACTION_META_MOCK, id: 2 };
-      const incomingTransaction2 = { ...TRANSACTION_META_MOCK, id: 3 };
+    it('ignores new transactions if hash matches existing incoming transaction', async function () {
+      const existingTransaction = {
+        ...TRANSACTION_META_MOCK,
+        id: 1,
+        type: TransactionType.incoming,
+      };
+      const existingTransaction2 = {
+        ...TRANSACTION_META_MOCK,
+        id: 1,
+        type: TransactionType.simpleSend,
+      };
+
+      const incomingTransaction1 = {
+        ...TRANSACTION_META_MOCK,
+        id: 2,
+        type: TransactionType.incoming,
+      };
+      const simpleSendTransaction2 = {
+        ...TRANSACTION_META_MOCK,
+        id: 3,
+        type: TransactionType.simpleSend,
+      };
 
       txController.store.getState().transactions = {
         [existingTransaction.id]: existingTransaction,
+        [existingTransaction2.id]: existingTransaction2,
       };
 
       await incomingTransactionHelperEventMock.firstCall.args[1]({
-        added: [incomingTransaction1, incomingTransaction2],
+        added: [incomingTransaction1, simpleSendTransaction2],
         updated: [],
       });
 
       assert.deepEqual(txController.store.getState().transactions, {
         [existingTransaction.id]: existingTransaction,
+        [existingTransaction2.id]: existingTransaction2,
+        [incomingTransaction1.id]: incomingTransaction1,
+        [simpleSendTransaction2.id]: simpleSendTransaction2,
```

Comment:
We should only have one of the new incoming transactions added as one should have the same hash but a different type so gets through, where as the other should be filtered as it has the same hash and the existing was incoming?